### PR TITLE
Feature/auto proposal period

### PR DESCRIPTION
**Pull request by martinyde**
https://jira.itkdev.dk/browse/FORSLAG-36

- Change add proposal form title field type
- Change display of sign out container.
- Add hook_presave for altering field values before saving a citizen_proposal

### CODE REVIEW

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 1]
```diff
@@ -155,6 +159,27 @@ public function proposalPageAttachments(&$page): void {
     }
   }
 
+  /**
+   * Implements hook_entity_presave().
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:165]: This “Implements” declaration does not match the one in hoeringsportal_citizen_proposal.module.

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 2]
```diff
@@ -155,6 +159,27 @@ public function proposalPageAttachments(&$page): void {
...
+   */
+  public function proposalEntityPresave(EntityInterface $entity): void {
+    if ('citizen_proposal' !== $entity->bundle()) {
+      return;
+    }
+    $proposalOriginal = $entity->original;
+    // Allow changing this value in settings.php $settings['sugggestion_period'] .
+    $periodLength = Settings::get('suggestion_period_length', self::SUGGESTION_PERIOD_LENGTH);
+    // If content is published.
+    if (1 === (int)$entity->status->value && 0 === (int)$proposalOriginal->status->value) {
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php]: You could assert that `$entity` is a `NodeInterface` here and use `$entity->isPublished()`.

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 3]
```diff
@@ -155,6 +159,27 @@ public function proposalPageAttachments(&$page): void {
...
+      // Set suggestion period.
+      $entity->set('field_vote_start', DrupalDateTime::createFromFormat('U', strtotime('now'))->format('Y-m-d\TH:i:s'));
+      $entity->set('field_vote_end', DrupalDateTime::createFromFormat('U', strtotime($periodLength))->format('Y-m-d\TH:i:s'));
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:181]: Use `new DrupalDateTime()` (implicit “now”) and `new DrupalDateTime($periodLength)` here.

Do we really have to use a string to set the value of a DateTime field? If so, can we use `DrupalDateTime::FORMAT` as the format?

**Review by rimi-itk — CHANGES_REQUESTED** [review]

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php (hunk 2 above)
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php]: Changed this to use isPublished method

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php (hunk 3 above)
Comment (martinyde) [reply web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php]: Changed this. I believe it has to be a string yes. 
DrupalDateTime::FORMAT does not work but DateTimeItemInterface::DATETIME_STORAGE_FORMAT does. I'm sure there is a reasoning behind that. I just don't see it..

File: web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php [hunk 4]
```diff
@@ -160,23 +163,29 @@ public function proposalPageAttachments(&$page): void {
...
-      $entity->set('field_vote_start', DrupalDateTime::createFromFormat('U', strtotime('now'))->format('Y-m-d\TH:i:s'));
-      $entity->set('field_vote_end', DrupalDateTime::createFromFormat('U', strtotime($periodLength))->format('Y-m-d\TH:i:s'));
+    // Allow changing this value in settings.php.
+    $periodLength = Settings::get('proposal_period_length', self::PROPOSAL_PERIOD_LENGTH);
+
+    $start = new DrupalDateTime();
+    $end = new DrupalDateTime($periodLength);
+    $storageTimezone = new \DateTimeZone(DateTimeItemInterface::STORAGE_TIMEZONE);
+
+    // If content is being published in this node->save() action.
```
Comment (rimi-itk) [inline web/modules/custom/hoeringsportal_citizen_proposal/src/Helper/Helper.php:183]: 🎉

**Review by rimi-itk — APPROVED** [review]

**Pull request merged**